binding/angular/projects/rhino-angular/src/lib/rhino.service.ts

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright 2022 Picovoice Inc.
+
+  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+  file accompanying this source.
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+import { Injectable, OnDestroy } from '@angular/core';
+import { Subject } from 'rxjs';
+
+import { WebVoiceProcessor } from '@picovoice/web-voice-processor';
+import {
+  InferenceCallback,
+  RhinoContext,
+  RhinoOptions,
+  RhinoInference,
+  RhinoModel,
+  RhinoWorker,
+} from '@picovoice/rhino-web';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RhinoService implements OnDestroy {
+  public inference$: Subject<RhinoInference> = new Subject<RhinoInference>();
+
+  public contextInfo$: Subject<string | null> = new Subject<string | null>();
+  public isLoaded$: Subject<boolean> = new Subject<boolean>();
+  public isListening$: Subject<boolean> = new Subject<boolean>();
+  public error$: Subject<Error | string | null> = new Subject<Error | string | null>();
+
+  private rhino: RhinoWorker | null = null;
+
+  constructor() {}
+
+  public async init(
+    accessKey: string,
+    context: RhinoContext,
+    model: RhinoModel,
+    options: RhinoOptions = {}
+  ): Promise<void> {
+    if (options.processErrorCallback) {
+      console.warn('\'processErrorCallback\' is only supported in the Rhino Web SDK. ' +
+       'Use the \'error\' state to monitor for errors in the Angular SDK.');
+    }
+
+    try {
+      if (!this.rhino) {
+        this.rhino = await RhinoWorker.create(
+          accessKey,
+          context,
+          this.inferenceCallback,
+          model,
+          { processErrorCallback: this.errorCallback }
```

Comment:
this ignores the other options. You need to unwrap options and replace `processErrorCallback`

Reply:
oops :)